### PR TITLE
Change Dodge the Creeps to be more consistent with the demo projects

### PR DESCRIPTION
**Pull request by aaronfranke**
Over time the docs and demos for Dodge the Creeps have drifted apart, so now I'm making an effort to unify them. With this PR, all the code examples are the same as the demo as of https://github.com/godotengine/godot-demo-projects/pull/531. This includes many bug fixes, enhancements, style changes, usage of best practices, and large C#-specific changes.

I haven't reviewed the images yet (most are probably fine), but I think this is all (or almost all) of the code changes.

### CODE REVIEW

**Review by NathanLovato — APPROVED** [review]

**Comment by NathanLovato** [comment]
Looking good to me, but I have barely used C# in Godot and would appreciate another pair of eyes for review.

File: getting_started/step_by_step/your_first_game.rst
Comment (akien-mga) [inline getting_started/step_by_step/your_first_game.rst:180]: GDScript doesn't have private variables per se, so this might be a bit misleading.

"Your first game" is also supposed to be simple, personally I'm not sure whether introducing users to public/private access at this stage is really useful.

Good practices are good but we should be careful with the simpler demos not to increase the overhead needlessly for beginners (that's why we don't recommend using type hints in simple demos for example).

Comment (NathanLovato) [reply getting_started/step_by_step/your_first_game.rst:180]: In our tuts we call these pseudo-private.

In the docs, I'd assume people have watched a CS intro course or have programming foundations; we'll link to some free resources as part of the getting started rewrite. Other pages already talk about programming patterns or classes, and there's not really a way around it considering Godot is an object-oriented engine, even on the user's side.

Here, the properties could well stay public. I don't know what's best.

Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst:180]: I agree with @akien-mga, this is introducing a whole set of concepts that are really of no help in making this program work. It only adds to the cognitive load - if you understand public/private, it's misleading, and if you don't, it's just confusing.

Comment (NathanLovato) [reply getting_started/step_by_step/your_first_game.rst:180]: I imagine the demo in godot-demo-projects was updated to be like that. So it's not you who changed the code, @cbscribe, right? I took it as you had updated the code or someone had consulted you before changing it.

Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst:180]: No, I haven't done any updates to the demo project. I did see there were changes to the C# version being made recently, but I didn't review them. Maybe it was slipped in during those?

Comment (NathanLovato) [reply getting_started/step_by_step/your_first_game.rst:180]: The changes were also applied to the GDScript version, yes: https://github.com/godotengine/godot-demo-projects/commit/dd11042d95d0cb996b815818d2766b35ad211e8e

Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:180]: It's easy to undo the changes to the demo projects, we can change these back to public.

**Comment by NathanLovato** [comment]
Thanks for making the variables public again. @cbscribe could you check the changes and tell us if they're fine to you?

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:242]: This seems like overexplaining for this game. Let's get them working with default input actions before talking about custom ones. This paragraph just implies you can do it but then leaves the_reader still wondering how.

Using the default actions (and these should be consistently referred to as "input actions") is what we want them doing, not an alternative choice.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:256]: I don't like this at all. The original code was explicit and clear. Each input is shown to map to a direction. This now gets into something called "action strength" which is not at all apparent what it means.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:503]: This one doesn't need to be deferred, because it's not called during the physics callback.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:726]: This change violates the GDScript style guide:

> Also use PascalCase when loading a class into a constant or a variable

In addition, `export` is not a function - removing the space before the export type hint makes it seem so.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:847]: What is the point of this?

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:851]: We're instancing a `mob` - why does this variable need to be expanded to `mob_instance`?

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:854]: This change is entirely unnecessary. Let's not get into the TAU vs PI argument.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:864]: This results in a much longer and harder to read line. We originally had it written like that in an early version of the game, and this led to additional confusion, which was why it was broken into multiple lines.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:860]: As above.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [inline getting_started/step_by_step/your_first_game.rst:1222]: Why move this to `new_game()`?

**Review by cbscribe — COMMENTED** [review]
I've added several review comments. In my opinion, most of these changes are unnecessary, if not outright negative.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:726]: Scenes are neither classes nor constants, and in all other instances of the demos exported PackedScenes are using snake_case so it would have to be changed there too.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:851]: For clarity's sake, but we can change this back.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:256]: This code is actually superior, since it properly supports joysticks, and it's also shorter. It seems clear to me, it says that the X velocity is equivalent to the right button minus the left button.

**Comment by NathanLovato** [comment]
Seeing all the review comments, I should've paid more attention to the PR. They all make sense.

So @cbscribe would you recommend to close the PR and revert changes on the demo? Seems to me like it's the way to go here.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst:726]: Well, we could discuss changing the style guide, but either way this code and the style guide should match.

**Comment by aaronfranke** [comment]
There are still things that can be improved and inconsistencies with the demos, I can amend the PR and change the demos. If you reverted the most recent PR in the demos, they would still not be consistent with the docs (because they weren't before).

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:1222]: This way you can see the enemies after you lose.

File: getting_started/step_by_step/your_first_game.rst
Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst:256]: I get it, but it's is not superior if the user is using default input actions (which they should be in this *beginner* demo).

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:726]: Actually, there is not a single example of `export(PackedScene)` in the GDScript style guide, so there's nothing that says this should be capitalized.

Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst:726]: @NathanLovato has this changed at some point? I believe it used to be included in this section:

>Also use PascalCase when loading a class into a constant or a variable:

> const Weapon = preload("res://weapon.gd")

There's no explicit PackedScene example in the guide now.

I'm fine either way, but this is a debate for the style guide, not for this PR. Whatever is preferred should be added there and followed.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:503]: Yes, you're right. This slipped in with a change to the demos and I missed it.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:847]: This same node is referenced 2 other times in the script below these lines, so this means that we don't need to get it thrice.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst:746]: Another interesting thing is that without the `scene` suffix and without the `instance` suffix, we lose the ability to explicitly refer to everything in C#. `mob` would be the instance, `Mob` is the class, so whether we went with a `mob` field or a `Mob` property we would be masking something somewhere.

File: getting_started/step_by_step/your_first_game.rst
Comment (aaronfranke) [reply getting_started/step_by_step/your_first_game.rst]: Good point, I changed this to be split into two lines, but I think it's better to use a local field rather than storing the temporary value inside of `linear_velocity`.

**Comment by NathanLovato** [comment]
Hey @aaronfranke , you should've waited for @cbscribe to check this out. This tutorial is being maintained by Chris, so the least we can do is ask him before making changes. Even it it takes time to get review. When he's too busy or misses the message, you can ping him on Discord.

**Comment by aaronfranke** [comment]
@NathanLovato I addressed his above changes and waited 10 days. If he finds any further issues, he is welcome to make a follow-up PR, or to yell at me and I will make a follow-up PR.

**Comment by NathanLovato** [comment]
It would've taken you 10 seconds to ping the reviewer again. Did you even try to contact him before merging? If not, please do so next time.

**Pull request merged**
